homeassistant/components/sense/__init__.py

```diff
@@ -69,16 +75,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     gateway.rate_limit = ACTIVE_UPDATE_RATE
 
     try:
-        await gateway.authenticate(email, password)
-    except SenseAuthenticationException:
-        _LOGGER.error("Could not authenticate with sense server")
-        return False
-    except SENSE_TIMEOUT_EXCEPTIONS as err:
-        raise ConfigEntryNotReady(
-            str(err) or "Timed out during authentication"
-        ) from err
-    except SENSE_EXCEPTIONS as err:
-        raise ConfigEntryNotReady(str(err) or "Error during authentication") from err
+        gateway.load_auth(access_token, user_id, monitor_id)
+        await gateway.get_monitor_data()
+    except (SenseAuthenticationException, SenseMFARequiredException) as err:
+        _LOGGER.warning("Sense authentication expired")
+        raise ConfigEntryAuthFailed(err) from err
 
     sense_devices_data = SenseDevicesData()
     try:
```

Comment:
The `trends_coordinator` `update_method` should also needs to be wrapped so it raises `ConfigEntryAuthFailed` from `(SenseAuthenticationException, SenseMFARequiredException)`

Reply:
I'm not too familiar with how the coordinator calls the update method - is there something that will handle the `ConfigEntryAuthFailed` exception?